Src/WitsmlExplorer.Frontend/components/QueryEditorWidgetUtils.tsx

```diff
@@ -0,0 +1,113 @@
+import { Icon } from "@equinor/eds-core-react";
+import { createRoot } from "react-dom/client";
+import styled from "styled-components";
+import { getTag } from "./QueryEditorUtils";
+
+/**
+ * Handles the click event for opening a tag in the QueryEditor.
+ * Replaces the self-closing tag with an opening and closing tag and moves the cursor between the tags.
+ * @param editor The Ace editor instance.
+ * @param row The row number of the tag to open.
+ */
+const onClickOpenTag = (editor: any, row: number) => {
+  const session = editor.getSession();
+  const lineText = session.getLine(row);
+  const tag = getTag(lineText);
+  const newText = `${lineText.replace(/(\s*)\/>/, ">")}</${tag ?? ""}>`;
+  session.replace({ start: { row, column: 0 }, end: { row, column: lineText.length } }, newText);
```

Comment:
When I tried to fix the types, the `session.replace()` method complained. Another method in [ace-builds/ace.d.ts](https://github.com/ajaxorg/ace-builds/blob/master/ace.d.ts) may be more suitable. We are now working with `Ace.EditSession`.

Reply:
We had a discussion and decided to skip this task because it was complicated due to issues with the package.